assets/global.js

```diff
@@ -533,10 +539,16 @@ class VariantSelects extends HTMLElement {
     const newMedia = document.querySelector(
       `[data-media-id="${this.dataset.section}-${this.currentVariant.featured_media.id}"]`
     );
+    const modalContent = document.querySelector(`#ProductModal-${this.dataset.section}`);
+    const newMediaModal = modalContent.querySelector( `[data-media-id="${this.currentVariant.featured_media.id}"]`);
+
     if (!newMedia) return;
     const parent = newMedia.parentElement;
+    if (parent.firstChild == newMedia) return;
+    modalContent.prepend(newMediaModal);
     parent.prepend(newMedia);
-    window.setTimeout(() => { parent.scroll(0, 0) });
+    parent.dispatchEvent(window.preventHeaderReveal);
```

Comment:
No, I just wanted to make sure it would be dispatch and could be grabbed from the sticky header. 
My idea was that you could reuse `dispatchEvent(window.preventHeaderReveal);` anywhere else it might be needed in the theme (if it's ever something that we need to call elsewhere) rather than declare new events every time. But if you say it's okay to do so, then I can change it 👌